### PR TITLE
[O11y][Apache Spark] Add metric_type for executor data stream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

- This PR adds metric_type mapping for the fields of the executor data stream.
- To verify the metric_type, please refer to this [file](https://github.com/elastic/integrations/blob/main/packages/apache_spark/data_stream/executor/agent/stream/stream.yml.hbs#L9-L269) mbeans type that is used for data collection

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

- Relates #7786 
- Relates #7408

### CODE REVIEW

**Review by aliabbas-elastic — APPROVED** [review]
LGTM

**Comment by agithomas** [comment]
@harnish-elastic , similar to https://github.com/elastic/integrations/pull/7794,is there a reference to metric_type mapping?

**Comment by harnish-elastic** [comment]
> @harnish-elastic , similar to #7794 there a reference to metric_type mapping?

@agithomas Since the MBean list is too large to mention in the PR description, I have attached the link of the metric_type in the [description](https://github.com/elastic/integrations/pull/7831#:~:text=To%20verify%20the%20metric_type) itself

File: packages/apache_spark/data_stream/executor/fields/fields.yml [hunk 1]
```diff
@@ -214,79 +260,100 @@
               fields:
                 - name: direct_memory
                   type: long
+                  metric_type: gauge
                   description: Amount of direct memory used by the shuffle client.
                 - name: heap_memory
                   type: long
+                  metric_type: counter
```
Comment (agithomas) [inline packages/apache_spark/data_stream/executor/fields/fields.yml:267]: Please recheck this 

```
  - mbean: 'metrics:name=*.*.ExternalShuffle.shuffle-client.usedHeapMemory,type=gauges'
    attributes:
    - attr: Value
      field: executor.shuffle.client.used.heap_memory
```

**Review by agithomas — APPROVED** [review]
LGTM!

File: packages/apache_spark/data_stream/executor/fields/fields.yml (hunk 1 above)
Comment (harnish-elastic) [reply packages/apache_spark/data_stream/executor/fields/fields.yml]: Updated, thank you!

**Pull request merged**